### PR TITLE
[WIP] Project update sync assets (round 2)

### PR DESCRIPTION
**Pull request by john-westcott-iv**
##### SUMMARY

This PR replaces PR #7099

This is an initial cut as having parts of Tower being controlled by code. A project update of an SCM based project will not attempt to create job templates and workflow job templates from files within the source. The current proposed files are:
assets/job_templates.yml
assets/workflow_job_template_nodes.yml
assets/workflow_job_templates.yml

This PR (currently):
* Adds a "Sync Assets" checkbox to the project UI screen. If enabled, during a manually invoked project sync the project_update.yml playbook will now include a role which will read these files and then leverage the awx.awx collection to build assets within tower.
* Adds an export endpoint to project API (still needs work).
* Adds a PROJECT_SYNC_ACCESS_TOKEN_EXPIRE_SECONDS setting
* Modifies UI for project export job types

Related
#526
#578

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API
 - UI

##### AWX VERSION

```
16.0.0
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by shanemcd** [comment]
Like we talked about earlier, I would be interested in seeing if we can associate a Tower credential to the project and use that instead of obtaining an oauth token behind the scenes.

**Comment by john-westcott-iv** [comment]
@ryanpetrello @wenottingham Should we allow for relaunch on a project export. Keeping in mind that an export takes a commit message and a list of job template IDs. On a failure I suppose it would make sense. However, on a successful job export would we want people to be able to relaunch with the same commit message?

**Comment by john-westcott-iv** [comment]
> Like we talked about earlier, I would be interested in seeing if we can associate a Tower credential to the project and use that instead of obtaining an oauth token behind the scenes.

@shanemcd @wenottingham If we use a Tower credential type for the project export it has a field called Ansible Tower Hostname (thats required). I feel like I should respect this when making the connection to Tower (right now I use TOWER_URL_BASE setting). This could lead to awkwardness on an export if someone does not populate that correctly. Any thoughts on this?

**Comment by john-westcott-iv** [comment]
recheck

File: awx/ui_next/src/screens/Project/shared/ProjectSubForms/SharedFields.jsx [hunk 1]
```diff
@@ -93,6 +93,15 @@ export const ScmTypeOptions = withI18n()(
                   revision of the project prior to starting the job.`
             )}
           />
+          <CheckboxField
+            id="option-sync_assets"
```
Comment (unlikelyzero) [inline awx/ui_next/src/screens/Project/shared/ProjectSubForms/SharedFields.jsx:97]: do we want underscores here? @mabashian

**Comment by unlikelyzero** [comment]
@john-westcott-iv , i'll take a stab at testing this from the ui-side when you're ready for review

**Comment by shanemcd** [comment]
Holy conflicts. Please let me know if you still want this.

**Comment by john-westcott-iv** [comment]
I def sill want it but didn't have the time to put it in before the 4.0 release so this will have to be completely refactored later. Closing for now.

**Pull request closed without merging**